happi/cli.py

```diff
@@ -396,21 +396,48 @@ def edit(ctx, name: str, edits: list[str]):
 
 
 @happi_cli.command()
-@click.argument('item_names', nargs=-1)
+@click.option('--glob/--regex', 'use_glob', default=True,
+              help='use glob style (default) or regex style search terms. '
+              r'Regex requires backslashes to be escaped (eg. at\\d.\\d)')
+@click.argument('search_criteria', nargs=-1)
 @click.pass_context
-def load(ctx, item_names: list[str]):
-    """Open IPython terminal with ITEM_NAMES loaded."""
+def load(
+    ctx: click.Context,
+    use_glob: bool,
+    search_criteria: list[str]
+):
+    term = []
+    results = set()
+
+    for item in search_criteria:
+        term.append(item)
+
+        final_results = search_parser(
+            client=get_happi_client_from_config(ctx.obj),
+            use_glob=use_glob,
+            search_criteria=term,
```

Comment:
My alternate nitpick suggestion is to do away with `term` and simplify the logic slightly for readability
```suggestion
            search_criteria=[item],
```